app/lib/pdf_filler/az321_pdf.rb

```diff
@@ -0,0 +1,58 @@
+class PdfFiller::Az321Pdf
+  include PdfHelper
+
+  def source_pdf_name
+    "az321-TY2023"
+  end
+
+  def initialize(submission)
+    @submission = submission
+
+    builder = StateFile::StateInformationService.submission_builder_class(:az)
+    @xml_document = builder.new(submission).document
+  end
+
+  def hash_for_pdf
+    answers = {
+      "TP_Name" => name('Primary'),
+      "TP_SSN" => @xml_document.at('Primary TaxpayerSSN')&.text,
+      "Spouse_Name" => name('Secondary'),
+      "Spouse_SSN" => @xml_document.at('Secondary TaxpayerSSN')&.text,
+      "4h" => @xml_document.at('ContTotalCharityAmt')&.text,
+      "4" => @xml_document.at('TotalCharityAmtContSheet')&.text,
+      "5" => @xml_document.at('TotalCharityAmt')&.text,
+      "11" => @xml_document.at('AddCurYrCrAmtTotCshCont')&.text,
+      "12" => @xml_document.at('TxPyrsStatus')&.text,
+      "13" => @xml_document.at('TotCshContrFostrChrty')&.text,
+      "20" => @xml_document.at('CurrentYrCr')&.text,
+      "22" => @xml_document.at('TotalAvailCr')&.text,
+    }
+
+    @submission.data_source.az321_contributions.each_with_index do |contribution, index|
+      if index <= 2
+        prefix = index + 1
+      else
+        letter = ('a'..'g').to_a[index - 3]
+        prefix = "4#{letter}"
```

Comment:
I don't know if we keep the user from entering more than 10 contributions elsewhere in the app, but for index >= 10 letter will be nil and prefix will just be "4". We might want to make this loop over a slice like az321_contributions[0..9] to make sure this doesn't happen (even if it's very unlikely)